### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by st--**
Minor updates:
* remove maintainers who stepped down
* add link to https://github.com/tgcsaba/GPSig

### CODE REVIEW

**Review by vdutor — COMMENTED** [review]
@st-- What is the current state of this PR? Have people that are being removed from the list been notified and are they happy with this? @alexisboukouvalas and @joelberkeley-secondmind

**Comment by joelberkeley** [comment]
makes sense to me

**Review by vdutor — APPROVED** [review]
LGTM - please see two minor comments.

File: README.md [hunk 1]
```diff
@@ -13,14 +13,35 @@
...
+* [Citing GPflow](#citing-gpflow)
+
+
+## What does GPflow do?
+
+GPflow is a package for building Gaussian process models in Python.
+It implements modern Gaussian process inference for composable kernels and likelihoods.
+The [online documentation (develop)](http://gpflow.readthedocs.io/en/develop/)/[(master)](http://gpflow.readthedocs.io/en/master/) contains more details.
+
+GPflow 2.1 builds on [TensorFlow 2.2+](http://www.tensorflow.org) and [TensorFlow Probability](https://www.tensorflow.org/probability/) for running computations, which allows fast execution on GPUs.
```
[the remark this answers is not in the document]
Comment (st--) [reply README.md:37]: I was thinking of just updating the 2.1 to 2.2 once that's released :thinking:

File: README.md [hunk 2]
```diff
@@ -13,14 +13,35 @@
...
+   * [Contributing](#contributing)
+   * [Projects using GPflow](#projects-using-gpflow)
+* [Version Compatibility](#version-compatibility)
+   * [TensorFlow 1.x and GPflow 1.x](#tensorflow-1x-and-gpflow-1x)
+* [Citing GPflow](#citing-gpflow)
+
+
+## What does GPflow do?
+
+GPflow is a package for building Gaussian process models in Python.
```
[the remark this answers is not in the document]
Comment (st--) [reply README.md:33]: It mentions in the next paragraph right below, do you think that's not obvious enough?

**Pull request merged**